dbt/adapters/trino/connections.py

```diff
@@ -64,7 +66,9 @@ def from_dict(cls, data: Any):
 
 
 class TrinoCredentials(Credentials, metaclass=ABCMeta):
+    DISABLE_PREPARED_STATEMENTS = DISABLE_PREPARED_STATEMENTS_DEFAULT
     _ALIASES = {"catalog": "database"}
+    _HTTP_CLIENT_PATCHED = False
```

Comment:
I think that http client patching is not in any way critical for this PR.

Can we get this change in a different PR to concentrate more thoroughly on it?

Reply:
Now that we added the flag for the prepared statements, that's true. It may however slightly increase the odds of the defaults failing.

I had discussed this issue earlier with @hovaesco, in the end I think Trino should play nicely with the Python ecosystem, so Trino should abide to any limits of Python (such as `http.client._LINE`). So ideally we get this fixed in Trino itself.

It's just that compared to Python scripting, the interface of dbt is not Python but the `dbt` command, so it's not easy for users to add this Python snippet and override the Python environment. So I thought it would be nice to ensure smooth operations of any existing large seed files and/or bumped up batch size that worked before but not anymore with the prepared statements.

Reply:
I'm not against adding this kind of patch.
I would rather discuss it more thoroughly (in a separate PR) before actually making it generally available.